modules/sdk-coin-xrp/src/xrp.ts

```diff
@@ -289,19 +287,27 @@ export class Xrp extends BaseCoin {
       throw new Error('missing required param txHex');
     }
     let transaction;
-    let txHex;
+    let txHex: string;
     try {
-      transaction = rippleBinaryCodec.decode(params.txHex);
-      txHex = params.txHex;
+      transaction = JSON.parse(params.txHex);
+      txHex = rippleBinaryCodec.encode(transaction);
     } catch (e) {
       try {
-        transaction = JSON.parse(params.txHex);
-        txHex = rippleBinaryCodec.encode(transaction);
+        transaction = rippleBinaryCodec.decode(params.txHex);
+        txHex = params.txHex;
       } catch (e) {
         throw new Error('txHex needs to be either hex or JSON string for XRP');
       }
     }
-    const id = computeBinaryTransactionHash(txHex);
+
+    let id: string;
+    // hashes ids are different for signed and unsigned tx
+    // first we try to get the hash id as if it is signed, will throw if its not
+    try {
+      id = rippleLib.hashes.hashSignedTx(txHex);
+    } catch (e) {
+      id = rippleLib.hashes.hashTx(txHex);
+    }
```

Comment:
NIT : 
rather than throwing an error and figuring out whether its signed or not, is there a way to detect if txHex is signed?

A quick look through the doc showed https://js.xrpl.org/functions/verifySignature.html , But not really sure if this helper function will work here or not though